components/dashboards-web-component/src/viewer/components/DashboardRenderer.jsx

```diff
@@ -160,6 +190,22 @@ export default class DashboardRenderer extends Component {
             DashboardThumbnail.saveDashboardThumbnail(dashboardName, dashboardContainerId);
         }
     }
+
+    onGoldenLayoutComponentAddEvent(component) {
+        const exportButton = document.createElement('i');
+        exportButton.title = 'Generate Report';
+        exportButton.className = 'fw fw-pdf widget-report-generation-button';
+        exportButton.addEventListener('click', () => {
+            this.setState({ widget: component.element[0] });
```

Comment:
Combine these `setState` calls into a single one.

Also, set these `state` properties are not initialized in the constructor. You should define/initialize all the properties in the `state` in the constructor. Applies for other places as well.

Reply:
Done